### DISCUSSION TITLE
Error position in decimal instead of hex

### DISCUSSION THREAD

**Discussion by ckrenslehner** [jam1garner/binrw#296] [Q&A]
Hi there! Is there a specific reason why the position of the error is printed as hex?
Example output here:
![grafik](https://github.com/user-attachments/assets/2445a81a-7754-46df-aafc-763936593ab7)

I wonder how I could change this to print in decimal (so 10 instead of 0xa)?

Greetings and thanks, Christian

**Comment by ckrenslehner** [comment]
I interpreted the output wrong sorry for that!

**Comment by ckrenslehner** [comment]
Ok sorry. 0x10 really is the error position.

Here is a more complete output

```
 ╺━━━━━━━━━━━━━━━━━━━━┅ Backtrace ┅━━━━━━━━━━━━━━━━━━━━╸

 0: Error: incomplete utf-8 byte sequence from index 5 at 0xa
           While parsing field 'text' in BleAndWifiInfo
     at datapoint-processor\src\parser\datapoint_values\common_types.rs:192
 1: While parsing field 'self_0' in Datapoint::ScannedBleInfo
     at datapoint-processor\src\parser.rs:29

 ╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸
 ```
 
 I would like to have an output like ` 0: Error: incomplete utf-8 byte sequence from index 5 at position 10`

**Comment by csnover** [comment]
binrw errors are structured, so you can walk them yourself and print error messages however you want. There is no way to change just the format of the stream position in the included implementations of `Debug`/`Display`.
